src/coprocessor/dag/rpn_expr/types/expr_builder.rs

```diff
@@ -278,6 +296,25 @@ where
     // Map pb func to `RpnFnMeta`.
     let func = fn_mapper(tree_node.get_sig(), tree_node.get_children())?;
     let args = tree_node.take_children().into_vec();
+
+    // Only Int/Real/Duration/Decimal/Bytes/Json will be decoded
+    let datums = datum::decode(&mut tree_node.get_val())?;
+    let mut implicit_args = Vec::with_capacity(datums.len());
+    for d in datums {
+        let arg = match d {
+            Datum::I64(n) => ScalarValue::Int(Some(n)),
+            Datum::U64(n) => ScalarValue::Int(Some(n as i64)),
+            Datum::F64(n) => ScalarValue::Real(Some(Real::new(n).unwrap())),
+            Datum::Dur(dur) => ScalarValue::Duration(Some(dur)),
+            Datum::Bytes(bytes) => ScalarValue::Bytes(Some(bytes)),
+            Datum::Dec(dec) => ScalarValue::Decimal(Some(dec)),
+            Datum::Json(json) => ScalarValue::Json(Some(json)),
+            Datum::Null => return Err(box_err!("push down datum should be NOT NULL")),
+            _ => unreachable!(),
```

Comment:
How about removing the NULL match arm and write `_ => return Err(box_err!("Unsupported push down datum kind ...."))` instead? In this way, if we add a new type to datum later, we don't need to change the code here. It will simply returning errors instead of resulting in panics.